src/js/Content/Features/Community/ProfileHome/FProfileStatus.js

```diff
@@ -0,0 +1,109 @@
+import {HTML, Localization, SyncedStorage} from "../../../../modulesCore";
+import {Feature, RequestData, SteamId, User} from "../../../modulesContent";
+
+export default class FProfileStatus extends Feature {
+
+    checkPrerequisites() {
+        return !this.context.isPrivateProfile;
+    }
+
+    async apply() {
+        const statusNode = document.querySelector(".profile_in_game_header");
+        if (!statusNode) { return; }
+
+        const ingameNode = document.querySelector(".profile_in_game_name");
+        const profileSteamId = SteamId.getSteamId();
+        let lenderSteamId = null;
+
+        // The input needed to get appid only appears when logged in
+        const appidNode = document.querySelector("input[name=ingameAppID]");
+        const bInGame = ingameNode && appidNode;
+        if (bInGame) {
+            const appid = appidNode.value;
+            const game = ingameNode.textContent;
+            // FinGameStoreLink
+            HTML.inner(ingameNode,
+                `<a href="//store.steampowered.com/app/${appid}">
+                <span data-tooltip-text="${Localization.str.view_in_store}">${game}</span>
+            </a>`);
+
+            if (SyncedStorage.get("profile_shared_game")) {
+                const userToken = await User.getUserToken();
+                const {response} = await RequestData.getJson(`https://api.steampowered.com/IPlayerService/IsPlayingSharedGame/v1/?access_token=${userToken}&steamid=${profileSteamId}&appid_playing=${appid}`, {"credentials": "omit"}).catch(() => false);
+                lenderSteamId = (response || {}).lender_steamid; // falsy if request failed
+            }
+        }
+
+        let profile = null;
+        let lender = null;
+        const steamids = [];
+        // if in-game or option profile_accurate_status is disabled, we don't want to request accurate status
+        if (!bInGame && SyncedStorage.get("profile_accurate_status")) {
+            steamids.push(profileSteamId);
+        }
+        // lenderSteamId is falsy when profile_shared_game option is disabled
```

Comment:
Not worth a comment

Reply:
I was thinking the logic became a bit complex, hence my comment.